lib/sqlalchemy/dialects/postgresql/psycopg2.py

```diff
@@ -866,6 +866,12 @@ def is_disconnect(self, e, connection, cursor):
                 "SSL SYSCALL error: EOF detected",
                 "SSL SYSCALL error: Operation timed out",
                 "SSL SYSCALL error: Bad address",
+                # This can occur in OpenSSL 1 when an unexpected EOF occurs.
+                # https://www.openssl.org/docs/man1.1.1/man3/SSL_get_error.html#BUGS
+                # It may also occur in newer OpenSSL for a non-recoverable I/O
+                # error as a result of a system call that does not set 'errno'
+                # in libc.
+                "SSL SYSCALL error: Success",
```

Comment:
@zzzeek we start to have a lot of constants here.
We could probably cache the list.
We could also generate a regexp with these alternatives like so `(option 1)|(option 2)|...|(option n)`, to have a single check here

Reply:
Well this is an exception re-throw case so we don't need to optimize for finely tuned performance.   it's easy to read right now and unambiguous what messages we are catching and I like being in control of that.    if you want to put the list above at the class level and then generate it into a `|` regexp right after, feel free (does that even save on performance for a reg that large? does it add to startup time?)